### ISSUE TITLE
loss becoming 'nan' and accuracy dropping to 5% using tf.keras

### ISSUE THREAD

**Issue by anirbankonar123** [tensorflow/tensorflow#26543]
<em>Please make sure that this is a bug. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Google colab env
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary): !pip3 install -U tensorflow==2.0.0-alpha0
- TensorFlow version (use command below): 2.0.0-alpha0
- Python version: 3.6
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version:
- GPU model and memory: no GPU used

You can collect some of this information using our environment capture [script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with
python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"

"loss becoming 'nan' and accuracy dropping to 5% using tf.keras, with fashion-mnist, on 4th epoch (sometimes it happens on 7th / 8th epoch)

"loss should not become 'nan' and accuracy should never drop unexpectedly
Print output : 
Epoch #1	 Loss: 0.679400	Accuracy: 0.729167
Epoch #2	 Loss: 0.558087	Accuracy: 0.770833
Epoch #3	 Loss: 0.487591	Accuracy: 0.812500
Epoch #4	 Loss: 0.429859	Accuracy: 0.833333
**Epoch #5	 Loss: nan	        Accuracy: 0.052083**
[Fashion_mnist_with_keras_eager_and_tf_data.zip](https://github.com/tensorflow/tensorflow/files/2949581/Fashion_mnist_with_keras_eager_and_tf_data.zip)

**Code to reproduce the issue**
Code attached in jupyter notebook format

**Other info / logs**
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.

**Comment by jvishnuvardhan** [comment]
@anirbankonar123 I don't see any issue with your code. Please post this kind of support questions in stackoverflow. Thanks!
My output as follows
Epoch #1	 Loss: 0.634148	Accuracy: 0.739583
Epoch #2	 Loss: 0.524849	Accuracy: 0.770833
Epoch #3	 Loss: 0.454392	Accuracy: 0.833333
Epoch #4	 Loss: 0.390557	Accuracy: 0.885417
Epoch #5	 Loss: 0.346580	Accuracy: 0.885417
78.02320790290833

When I ran it again
Epoch #1	 Loss: 0.579839	Accuracy: 0.791667
Epoch #2	 Loss: 0.431964	Accuracy: 0.833333
Epoch #3	 Loss: 0.382009	Accuracy: 0.895833
Epoch #4	 Loss: 0.381249	Accuracy: 0.895833
Epoch #5	 Loss: 0.360117	Accuracy: 0.895833
81.38940477371216

As you see, accuracy is increasing each epoch. you could increase accuracy by following other tutorials on performance. Thanks

**Comment by anirbankonar123** [comment]
I got this on 8th epoch last time. I hv attached jupyter notebook also, it
is happening randomly at diff epochs. pls see my log here
https://colab.research.google.com/drive/1NmxW4FuBRH5-0t87YMl21Y-OKKRKCjun

There is something wrong here, loss shd never become nan, never seen this
in keras or Tf 1.x

Thanks

**Comment by jvishnuvardhan** [comment]
@anirbankonar123 Again, I ran your shared file. 
https://colab.sandbox.google.com/gist/jvishnuvardhan/becdfc18f1510f27bca54176048b3ba7/copy-of-fashion-mnist-with-keras-eager-and-tf-data.ipynb

Epoch #1	 Loss: 0.547747	Accuracy: 0.791667
Epoch #2	 Loss: 0.412555	Accuracy: 0.833333
Epoch #3	 Loss: 0.339921	Accuracy: 0.854167
Epoch #4	 Loss: 0.353694	Accuracy: 0.916667
Epoch #5	 Loss: 0.355032	Accuracy: 0.895833
222.52565932273865

Could you try running in other system or "reset all runtimes" in google colab and run it. Thanks!

**Comment by anirbankonar123** [comment]
As I mentioned, its not a reproducible issue, i got it initially on 7th
epoch, then didnt get it for a while, again got it twice today morning on
4th epoch. I generally reset all runtimes, its kind of mandatory for me as
I install fresh version of TF and need to make sure this is happpening
properly ! I understand its not coming now, for me too it was working
yesterday, but an issue all the while, as I didnt make a single code change
in this time. and its the same env as well...Thanks

**Comment by anirbankonar123** [comment]
Ran just now, after doing 'reset all runtimes' , installing tensorflow now
getting loss = nan on 3rd epoch :
https://colab.research.google.com/drive/1NmxW4FuBRH5-0t87YMl21Y-OKKRKCjun#scrollTo=kNgnUKPvgSCz

Please see here.
Thanks

**Comment by anirbankonar123** [comment]
Epoch #1 Loss: 0.549081 Accuracy: 0.718750 Epoch #2 Loss: 0.430481
Accuracy: 0.822917 Epoch #3 Loss: nan Accuracy: 0.166667 Epoch #4 Loss: nan
Accuracy: 0.166667 Epoch #5 Loss: nan Accuracy: 0.166667

**Comment by dksb** [comment]
@jvishnuvardhan Let's investigate this one further. There might be an environment/device specific issue.

**Comment by caisq** [comment]
cc myself

**Comment by jvishnuvardhan** [comment]
@anirbankonar123 Is this still an issue? Could you check with `tf-nightly-gpu-2.0-preview` and let us know how it progresses. I ran several times today, never saw `nan`. Thanks!
